### PR TITLE
Igor/mmd 1752/fix cert

### PR DESCRIPTION
**Pull request by rtviwe**

### CODE REVIEW

**Comment by rtviwe** [comment]
https://flutter-developer.medium.com/flutter-android-7-certificate-verify-failed-with-letsencrypt-ssl-cert-after-sept-30-2021-9cb5f672c090

File: logic/lib/src/data/api/client/client.dart [hunk 1]
```diff
@@ -1,13 +1,50 @@
...
 import 'package:mxc_logic/src/data/api/client/authentication.dart';
 
-import '../supernode/client_index.dart';
 import '../supernode/supernode.swagger.dart';
 import 'error_converter.dart';
 
 @internal
 class SupernodeClient extends ChopperClient {
+  // ignore: leading_newlines_in_multiline_strings
+  static const String isrgx1 = '''-----BEGIN CERTIFICATE-----
```
Comment (festelo) [inline logic/lib/src/data/api/client/client.dart:16]: move in different file pls

Comment (festelo) [inline logic/lib/src/data/api/client/client.dart:16]: move in different file pls

File: logic/lib/src/data/api/client/client.dart [hunk 2]
```diff
@@ -31,6 +69,27 @@ class SupernodeClient extends ChopperClient {
           errorConverter: ChopperErrorConverter(),
         );
 
+  static HttpClient _customHttpClient({String? cert}) {
```
Comment (festelo) [inline logic/lib/src/data/api/client/client.dart:72]: with this thing

File: logic/lib/src/data/api/client/client.dart [hunk 3]
```diff
@@ -31,6 +69,27 @@ class SupernodeClient extends ChopperClient {
...
+    final context = SecurityContext.defaultContext;
+    try {
+      if (cert != null) {
+        final bytes = utf8.encode(cert);
+        context.setTrustedCertificatesBytes(bytes);
+      }
+      log('createHttpClient() - cert added!');
+    } on TlsException catch (e) {
+      final fuck =
```
Comment (festelo) [inline logic/lib/src/data/api/client/client.dart:81]: not really clear variable name

File: logic/lib/src/data/api/client/client.dart [hunk 4]
```diff
@@ -31,6 +69,27 @@ class SupernodeClient extends ChopperClient {
...
+          e.osError?.message.contains('CERT_ALREADY_IN_HASH_TABLE') ?? false;
+      if (e.osError?.message != null && fuck) {
+        log('createHttpClient() - cert already trusted! Skipping.');
+      } else {
+        log('createHttpClient().setTrustedCertificateBytes EXCEPTION: $e');
+        rethrow;
+      }
+    } finally {}
```
Comment (festelo) [inline logic/lib/src/data/api/client/client.dart:89]: empty finally makes no sense

File: logic/lib/src/data/data.dart [hunk 5]
```diff
@@ -2,4 +2,5 @@ export 'api/api.dart';
 export 'cache/cache_field_manager.dart';
 export 'cache/cache_manager.dart';
 
-String sembastDbName = 'best.db';
+const sembastDbName = 'best.db';
+const sslUrl = 'https://valid-isrgrootx1.letsencrypt.org/';
```
Comment (festelo) [inline logic/lib/src/data/data.dart:6]: isn't clear what is ssl url and how it is used

**Review by festelo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**